README
        physical Package
    - Copyright 2003-2009 Spencer E. Olson

Much of the data for the units and constants listed in this header are
freely available on usenet servers.  The reference for the original
header is comp.lang.cpp/2004-10/2217 by E. Robert Tisdale (10/17/04).
The namespace layout is original to the posting by E. Robert Tisdale.

E. R. Tisdale <E.Robert.Tisdale_at_jpl.nasa.gov>.

There were a lot of other comments on usenet for improving the header from
Tisdale, including British spelling of various units and so on.  This library
is based on the unaltered version from E. R. Tisdale.


Since importing the file from E. R. Tisdale, I've extended this library heavily.
Various new units and constants have been added.  Most constants have been
checked and corrected according to CODATA from NIST:
http://physics.nist.gov/cuu/Constants/index.html

This software is released under the MIT license (see COPYING file).



ChangeLog
commit 520bae5a638704d50e66078bb9e961f5d3ce21f7
Author: Spencer E. Olson <olsonse@olsonse-laptop.(none)>
Date:   Wed Nov 4 14:36:15 2009 -0700

    added installation options

commit d63259c502e0a6ef74be20ec131882ab40a531fb
Author: Spencer E. Olson <olsonse@olsonse-laptop.(none)>
Date:   Tue Oct 27 13:57:48 2009 -0600

    added missing cstdlib include for std::abs(int)

commit 8f4dd1f4a714a99df6d2dbdc1f0e1aad6f176298
Author: Spencer E. Olson <olsonse@olsonse-laptop.(none)>
Date:   Sat Oct 24 21:13:10 2009 -0600

    added alias runtime.h header file for when using runtime::physical::*

commit da18f85649fc4310a2d45045b10148033199845e
Author: U-OW\olsonse <olsonse@ow.(none)>
Date:   Mon Oct 19 13:17:51 2009 -0700

    Fixed LATEX_PRINT and changed many BOOST_CHECK_EQUAL to BOOST_CHECK_CLOSE.
    
    I also changed the Jamfile import of /boost//unit_test_framework to be
    statically linked.  This seemed to be very helpful for using in Cygwin on
    windows.
    
    Some more unit tests for quantity printing were added.

commit da72662bf0733cabc4ec1d270fc231f209325908
Author: Spencer Olson <olsonse@asok.hpmi.afrl.af.mil>
Date:   Fri Oct 16 13:31:12 2009 -0600

    updated jam files to work with boost correctly. other small minor fixes

commit 3085a13551bbd8b14c3cef7441fe770018889b82
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Sat Oct 3 10:09:16 2009 +0000

    tweak Jamroot and update usage in testParer.cpp
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@845 e877f351-6dfb-0310-a246-97bd42f90100

commit 63e4ae24f0c7c3e3d85ba3f34cc1228c3918e446
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Tue Jun 23 17:07:27 2009 +0000

    version info changed for testInteractive
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@774 e877f351-6dfb-0310-a246-97bd42f90100

commit 1d45a055ab8900dede7fdb9454f7c54b176ab61a
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Tue Jun 23 17:02:57 2009 +0000

    added ability to add units in the boost::spirit grammar calculator
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@773 e877f351-6dfb-0310-a246-97bd42f90100

commit 83631b53708f5372b8957d430b61312a7515fdc3
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 14 18:31:26 2009 +0000

    Added changelog
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@764 e877f351-6dfb-0310-a246-97bd42f90100

commit 629ab9d07e7a85715be0f5e7e90afd1c9574b5b6
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 14 18:25:29 2009 +0000

    added author info on octave stuff
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@763 e877f351-6dfb-0310-a246-97bd42f90100

commit f087ac6e9e4ef5d8f56b4e59839c6dd26669750e
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 14 18:14:54 2009 +0000

    Tweaked template definitions to avoid unit system definitions begin fully defined and resulting in 'multiple definition' link errors
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@762 e877f351-6dfb-0310-a246-97bd42f90100

commit 22ba1630b1a3a7b64f9ee72f703ffe756af7061d
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 7 22:09:49 2009 +0000

    added cmake files.  I'm not sure how well they will work yet...
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@759 e877f351-6dfb-0310-a246-97bd42f90100

commit b8e1ed66b58c6030c6a9a651d62d0b428aa3ca3e
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 7 20:12:53 2009 +0000

    updated documentation a little.
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@757 e877f351-6dfb-0310-a246-97bd42f90100

commit e4fb69404883038137f2299d9ad2aba93752550a
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 7 20:05:20 2009 +0000

    fixed make_dim and exp_dim template meta functions.  All tests run (and will only a tiny exception pass).  Examples all compile
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@756 e877f351-6dfb-0310-a246-97bd42f90100

commit 5b2caaa0637f1b7f5cba77bdbce6b3815a06c0fc
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 7 08:34:17 2009 +0000

    "major update of physical library
    
    1.  unit systems have been implemented better.
        I currently have si, atomic, cgs, esu, emu, gaussian, heaviside_lorenzt.
        Both gaussian and heaviside_lorentz are still incomplete as I need to define
        charge for these two systems.
        cgs does not have charge defined and should not in the future since plain
        cgs does not necessarily define the electric/magnetic units.
    2.  templated conversion between systems has been implemented
    3.  "const Quantity" type of constants have been moved to
        physical::constant::si:: and templated counterparts that do automatic
        conversion into a desired unit system have been placed in
        physical::constant::
    
    
    
    git-svn-id: svn+ssh://sftp.itd.umich.edu/afs/umich.edu/user/o/l/olsonse/src/svn-repository/trunk/physical@755 e877f351-6dfb-0310-a246-97bd42f90100

commit 83815ec0350ef44c1aa599d7836d719899b69142
Author: olsonse <olsonse@e877f351-6dfb-0310-a246-97bd42f90100>
Date:   Thu May 7 07:05:10 2009 +0000

    added quantity<T>::getCoeff() which returns a constant reference to the coeff. Not sure this is really needed.
    

cxx/physical/element.h
/** \file
 * A set of data for elements from the periodic table.
 */

#if ( defined(PHYSICAL_DATA_FOR_RUNTIME) && \
     !defined(runtime_physical_element_h) ) || \
    (!defined(PHYSICAL_DATA_FOR_RUNTIME) && \
     !defined(physical_element_h) )

#  if defined(PHYSICAL_DATA_FOR_RUNTIME)
#    define runtime_physical_element_h
#  else
#    define physical_element_h
#  endif

#  if defined (PHYSICAL_DATA_FOR_RUNTIME)
namespace runtime {
#  endif

_OPEN_NAMESPACE(physical_element,physical) {
    // #### BEGIN OF ELEMENT CREATION #### #
    // FIXME:  finish filling out this periodic table data
    /* This data is taken from physics.nist.gov. */
#define _ELEMENT(name,sym,n,m,i) \
    _BEGIN_NAMESPACE(name) { \
        _QUANTITYn(number,    PHYSICAL_QUANTITY_COEFF_TYPE(n),"atomic number of "     #name); \
        _QUANTITYn(mass,        m * constant::si::amu, "atomic mass of "       #name); \
        _QUANTITYn(ionization,  i * constant::si::eV,  "ionization energy of " #name); \
    _END_NAMESPACE} \
    _ALIAS_NAMESPACE(sym, name);

    /** Periodic table information.
     * @see http://physics.nist.gov/PhysRefData/Elements/index.html
     */
    _BEGIN_NAMESPACE(element) {
        //       long-name    abrv.   n        mass(amu)  outer-electron ionization (eV)
        _ELEMENT(hydrogen,      H,    1,      1.00794,      13.5984 );
        _ELEMENT(helium,        He,   2,      4.002602,     24.5874 );
        _ELEMENT(lithium,       Li,   3,      6.941,         5.3917 );
        _ELEMENT(beryllium,     Be,   4,      9.012182,      9.3227 );
        _ELEMENT(boron,          B,   5,     10.812,         8.2980 );
        _ELEMENT(carbon,         C,   6,      12.0107,      11.2603 );
        _ELEMENT(nitrogen,       N,   7,      14.0067,      14.5341 );
        _ELEMENT(oxygen,         O,   8,      15.9994,      13.6181 );
        _ELEMENT(fluorine,       F,   9,      18.9984032,   17.4228 );
        _ELEMENT(neon,          Ne,  10,      20.1797,      21.5645 );
        _ELEMENT(sodium,        Na,  11,      22.989770,     5.1391 );
        _ELEMENT(magnesium,     Mg,  12,      24.3051,       7.6462 );
        _ELEMENT(aluminium,     Al,  13,      26.981538,     5.9858 );
        _ELEMENT(silicon,       Si,  14,      28.0855,       8.1517 );
        _ELEMENT(phosphorus,     P,  15,      30.973761,    10.4867 );
        _ELEMENT(sulfur,         S,  16,      32.065,       10.3600 );
        _ELEMENT(chlorine,      Cl,  17,      35.453,       12.9676 );
        _ELEMENT(argon,         Ar,  18,      39.948,       15.7596 );
        _ELEMENT(potassium,      K,  19,      39.0983,       4.3407 );
        _ELEMENT(calcium,       Ca,  20,      40.078,        6.1132 );
        _ELEMENT(scandium,      Sc,  21,      44.955911,     6.5615 );
        _ELEMENT(titanium,      Ti,  22,      47.867,        6.8281 );
        _ELEMENT(vanadium,       V,  23,      50.9415,       6.7462 );
        _ELEMENT(chromium,      Cr,  24,      51.9961,       6.7665 );
        _ELEMENT(manganese,     Mn,  25,      54.9380499,    7.4340 );
        _ELEMENT(iron,          Fe,  26,      55.845,        7.9024 );
        _ELEMENT(cobalt,        Co,  27,      58.9332009,    7.8810 );
        _ELEMENT(nickel,        Ni,  28,      58.6934,       7.6398 );
        _ELEMENT(copper,        Cu,  29,      63.546,        7.7264 );
        _ELEMENT(zinc,          Zn,  30,      65.409,        9.3942 );
        _ELEMENT(krypton,       Kr,  36,      83.798,       13.9996 );
        _ELEMENT(rubidium,      Rb,  37,      85.4678,       4.1771 );
        _ELEMENT(xenon,         Xe,  54,     131.293,       12.1298 );
        _ELEMENT(cesium,        Cs,  55,     132.90545,      3.8939 );
        _ELEMENT(mercury,       Hg,  80,     200.59,        10.4375 );
        _ELEMENT(francium,      Fr,  87,     223,            4.0727 );
    _END_NAMESPACE}
    _ALIAS_NAMESPACE(elements, element);
    // ####   END OF ELEMENT CREATION #### #
_CLOSE_NAMESPACE(physical_element)}
#undef _ELEMENT

#  if defined (PHYSICAL_DATA_FOR_RUNTIME)
}/* namespace runtime. */
#  endif

#endif // physical_element_h
